bin/import_swb.py

```diff
@@ -13,32 +17,88 @@
 import unicodedata
 import wave
 import codecs
+import tarfile
+#from util.downloader import maybe_download, SIMPLE_BAR
 
 from util.text import validate_label
+ARCHIVE_NAME = 'switchboard_word_alignments.tar.gz'
+ARCHIVE_URL = 'http://www.openslr.org/resources/5/'
+ARCHIVE_DIR_NAME = 'LDC97S62'
+LDC_DATASET = 'swb1_LDC97S62.tgz'
+
+import requests
+
+def download_file(folder, url):
+    # https://stackoverflow.com/a/16696317/738515
+    local_filename = url.split('/')[-1]
+    full_filename = os.path.join(folder, local_filename)
+    print(local_filename, full_filename) 
+    r = requests.get(url, stream=True)
+    with open(full_filename, 'wb') as f:
+        for chunk in r.iter_content(chunk_size=1024): 
+            if chunk: # filter out keep-alive new chunks
+                f.write(chunk)
+    return full_filename
+
+def maybe_download(archive_url, target_dir, ldc_dataset):
+    # If archive file does not exist, download it...
+    archive_path = os.path.join(target_dir, ldc_dataset)
+    ldc_path = archive_url+ldc_dataset
+    if not os.path.exists(target_dir):
+        print('No path "%s" - creating ...' % target_dir)
+        makedirs(target_dir)
+
+    if not os.path.exists(archive_path):
+        print('No archive "%s" - downloading...' % archive_path)
+        download_file(target_dir, ldc_path)
+    else:
+        print('Found archive "%s" - not downloading.' % archive_path)
+    return archive_path
+
 
 def _download_and_preprocess_data(data_dir):
-    data_dir = os.path.join(data_dir, "LDC97S62")
+    new_data_dir = os.path.join(data_dir, ARCHIVE_DIR_NAME)
+    target_dir = os.path.abspath(new_data_dir)
+    archive_path = os.path.abspath(os.path.join(data_dir, LDC_DATASET))
+
+    # Check swb1_LDC97S62.tgz then extract
+    assert(os.path.isfile(archive_path))
+    _extract(target_dir, archive_path)
+    
+    # Transcripts
+    transcripts_path = maybe_download(ARCHIVE_URL, target_dir, ARCHIVE_NAME)
```

Comment:
Nice! Downloading the ISIP corrected transcripts.

My one question is: How do the transcripts you refer to a http://www.openslr.org/resources/5/ differ from the official ISIP ones [here](https://www.isip.piconepress.com/projects/switchboard/)?

Could you make sure this downloads the newest transcripts and also add a comment where `ARCHIVE_NAME` is defined indicating which version of the ISIP transcript is being downloaded.

Reply:
They are identical it seems:

```
rob@rob-tpt470s:~/Downloads$ diff switchboard_word_alignments.tar.gz switchboard_word_alignments_openslr.tar.gz 
rob@rob-tpt470s:~/Downloads$ crc32 switchboard_word_alignments.tar.gz switchboard_word_alignments_openslr.tar.gz 
c0765d32	switchboard_word_alignments.tar.gz
c0765d32	switchboard_word_alignments_openslr.tar.gz
```

Reply:
As I need to built on this result now, I've opened a new issue #1747 for this last point on the ISIP transcript version, and will merge this code and fix #1747 in another PR.